tauri-app/src-tauri/src/wine_manager.rs

```diff
@@ -53,26 +53,55 @@ impl WineManager {
         .spawn().expect("Failed to init prefix").wait().expect("Failed to init prefix");
     }
 
-    pub fn launch_exe(&self, absolute_path: &str, args: &Vec<&str>) -> std::io::Result<()> {
+    pub fn launch_exe(&self, absolute_path: &str, vars: &Vec<(&str, &str)>, args: &Vec<&str>) -> std::io::Result<()> {
         let display = env::var("DISPLAY").unwrap_or_else(|_| ":0".to_string());
         let mut command = Command::new("wine");
-        command.env("WINEPREFIX", &self.prefix).env("DISPLAY", display);
         command.arg(absolute_path);
+
+        for (var, value) in vars {
+            command.env(var, value);
+        }
         for arg in args {
             command.arg(arg);
         }
 
-        command.spawn()?.wait()?;
+        command
+            .env("WINEPREFIX", &self.prefix)
+            .env("DISPLAY", display)
+            .spawn()?
+            .wait()?;
+
         Ok(info!("Launched {absolute_path}"))
     }
 
-    pub(crate) fn load_cd(&self, source_dir: &str, drive_letter: &str) {
-        let source_path = self.get_c_path(source_dir);
+    pub(crate) fn load_cd(&self, source_dir: &str, drive_letter: &str) -> io::Result<()> {
+        let source_dir = format!("../drive_c/{}", source_dir);
```

Comment:
I'd prefer we leave this as `get_c_path`, so that we don't have to handle quirks later in multiple places

Reply:
We can't. It will fail because it attemps to symlink to a folder `/var/data/wine/drive_c/` in the host's filesystem because of the way symlinks work.

If you run an `ls -la` on the `/var/wine/dosdevices` inside the flatpak you will see that the other drives are symlinked with a relative path as well.

Reply:
I was wrong about this and have made the requested changes.

I switched to using a relative path because of the issues I was have getting the CD to mount for first launch, and mistakenly thought that one of the reasons I got it working was because of the symlink behaviour.

The symlink is still valid when accessed from the sandbox, as you said it would be.